server/README-ECLIPSE.md

```diff
@@ -0,0 +1,40 @@
+Eclipse Installation
+--------------------
+
+The installation of the various plug-ins we use for development,
+code quality, etc., is straightforward, but is nowhere near as automated
+as we would like. To install an Eclipse distribution with all the required
+tools and get the project ready for development, do the following:
+
+  1. Download the Eclipse Oxygen installer for your platform from
+  http://www.eclipse.org/oxygen/.
+  2. Run the installer and install the "Eclipse IDE for Java Developers"
+  (the first option on the list of distributions to install).
```

Comment:
Yes, you do run `eclpse-inst`, and it surprises me that the Eclipse people don't have reasonable documentation about it.

Sadly, there's no way to get "security" with an Eclipse distribution at present - even if Eclipse was signed or properly hashed, most of the static checking plugins we use are not. I don't consider it a serious problem for the security of the software under development because we don't need to build it with Eclipse (and in particular, the continuous integration build will not be performed with Eclipse). It is, of course, a more serious problem in terms of being worried about what untrusted software might do to your machine.

In the future I'd like to explore different IDEs to see if we can find one with both a better security story around it and support for all the different tools we want to use. My guess is we can't.